src/transformers/pipelines.py

```diff
@@ -2318,7 +2318,7 @@ def _concat_inputs_history(self, inputs: List[List[int]], histories: List[Option
         max_len = max([len(item) for item in outputs])
         outputs = [output + [self.pad_token_id] * (max_len - len(output)) for output in outputs]
         outputs = BatchEncoding(
-            {"input_ids": outputs, "attention_mask": [1] * len(outputs)}, tensor_type=self.framework
+            {"input_ids": outputs, "attention_mask": [[1] * len(outputs)]}, tensor_type=self.framework,
```

Comment:
This is the only thing were the change of dim in BatchEncoding.convert_to_tensors` is breaking something, but in this case, it was a bit magical that the dimension was automatically added, so I don't think this is a serious failure.